controllers/pvc_storageclass_test.go

```diff
@@ -0,0 +1,194 @@
+				storageclass.NewAnnotationSetter(mgr.GetClient()),
+			})
+
+			mgr.GetWebhookServer().Register(webhookMutatePVCsPath, &webhook.Admission{Handler: pvcMutator})
+			Expect(err).NotTo(HaveOccurred(), "failed to setup controller")
+		}
+
+		Expect(k8sClient.Create(ctx, defStosSC)).Should(Succeed())
+		Expect(k8sClient.Create(ctx, stosSC)).Should(Succeed())
+		Expect(k8sClient.Create(ctx, notStosSC)).Should(Succeed())
+
+		go func() {
+			err := mgr.Start(ctx)
+			Expect(err).NotTo(HaveOccurred(), "failed to start manager")
+		}()
+
+		// Wait for manager to be ready.
+		time.Sleep(managerWaitDuration)
+	})
+
+	AfterEach(func() {
+		Expect(k8sClient.Delete(ctx, defStosSC)).Should(Succeed())
+		Expect(k8sClient.Delete(ctx, stosSC)).Should(Succeed())
+		Expect(k8sClient.Delete(ctx, notStosSC)).Should(Succeed())
+
+		cancel()
+	})
+}
+
+var _ = Describe("PVC StorageClass UID to annotation controller", func() {
+	// Define utility constants for object names and testing timeouts/durations
+	// and intervals.
+	const (
+		timeout  = time.Second * 1
+		duration = time.Second * 1
+		interval = time.Millisecond * 250
+	)
+
+	ctx := context.Background()
+
+	Context("When the PVC Mutator has no mutators", func() {
+		SetupPVCStorageClassAnnotationTestTest(ctx, false)
+
+		It("The pvc should be created", func() {
+			pvc := genPVC()
+
+			By("Creating the PVC")
+			Expect(k8sClient.Create(ctx, &pvc)).Should(Succeed())
+			defer func() {
+				Expect(k8sClient.Delete(ctx, &pvc)).Should(Succeed())
+			}()
+
+			By("Expecting the PVC to be unchanged")
+			Consistently(func() corev1.PersistentVolumeClaim {
+				got := corev1.PersistentVolumeClaim{}
+				err := k8sClient.Get(ctx, client.ObjectKeyFromObject(&pvc), &got)
+				if err != nil {
+					return corev1.PersistentVolumeClaim{}
+				}
+				return got
+			}, timeout, interval).Should(Equal(pvc))
+		})
+	})
+
+	Context("When the PVC Mutator has mutators", func() {
+		SetupPVCStorageClassAnnotationTestTest(ctx, true)
+
+		It("The pvc has default StorageClass", func() {
+			scName := deaultStorageClassName
+
+			pvc := genPVC()
+			pvc.Spec.StorageClassName = &scName
+
+			By("Creating the PVC")
+			Expect(k8sClient.Create(ctx, &pvc)).Should(Succeed())
+			defer func() {
+				Expect(k8sClient.Delete(ctx, &pvc)).Should(Succeed())
+			}()
+
+			persistedSC := storagev1.StorageClass{}
+			err := k8sClient.Get(ctx, client.ObjectKey{Name: scName}, &persistedSC)
+			Expect(err).NotTo(HaveOccurred(), "failed to fetch StorageClass")
+
+			By("Expecting StorageClass have configured as annotation")
+			Eventually(func() string {
+				var mutatedPVC corev1.PersistentVolumeClaim
+
+				err := k8sClient.Get(ctx, client.ObjectKeyFromObject(&pvc), &mutatedPVC)
+				if err != nil {
+					return ""
+				}
+
+				return mutatedPVC.Annotations[provisioner.StorageClassUUIDAnnotationKey]
+			}, timeout, interval).Should(Equal(string(persistedSC.UID)))
+		})
+	})
```

Comment:
Even though the unit tests cover this, I'd really like there to be test cases that create PVCs with different StorageClass provisioners & defaults and checks the end result. The encryption mutator is a good example.